src/com/google/enterprise/adaptor/documentum/DocumentumAdaptor.java

```diff
@@ -1570,9 +1570,14 @@ private boolean matchObjectPathsToDocId(DocId id, IDfSession dmSession,
     if (index == -1) {
       return false;
     }
+    String path = docIdPath.substring(0, index);
+    if (path.isEmpty()
+        && sysObject.getObjectId().toString()
+        .regionMatches(true, 0, String.format("%02x", DfId.DM_CABINET), 0, 2)) {
```

Comment:
The 0c expression could be a constant (DM_CABINET_TAG?). That would also allow us to indent this four more spaces for clarity, even if the tools or rules don't require that.